File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.java
import java.util.*;

class Solution {
    public int minimumPushes(String word) {
        // Count frequency of each character
        int[] freq = new int[26];
        for (char c : word.toCharArray()) {
            freq[c - 'a']++;
        }
        
        // Sort frequencies in descending order
        Integer[] indices = new Integer[26];
        for (int i = 0; i < 26; i++) {
            indices[i] = i;
        }
        Arrays.sort(indices, (a, b) -> freq[b] - freq[a]);
        
        int pushes = 0;
        for (int i = 0; i < 26; i++) {
            if (freq[indices[i]] == 0) break;
            pushes += freq[indices[i]] * (1 + i / 8);
        }
        
        return pushes;
    }
}

File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.c
#include <stdlib.h>
#include <string.h>

int compare(const void* a, const void* b, void* arg) {
    int* freq = (int*)arg;
    return freq[*(int*)b] - freq[*(int*)a];
}

int minimumPushes(char* word) {
    int freq[26] = {0};
    int len = strlen(word);
    
    // Count frequency of each character
    for (int i = 0; i < len; i++) {
        freq[word[i] - 'a']++;
    }
    
    // Create array of indices
    int indices[26];
    for (int i = 0; i < 26; i++) {
        indices[i] = i;
    }
    
    // Sort indices based on frequency
    qsort_r(indices, 26, sizeof(int), compare, freq);
    
    int pushes = 0;
    for (int i = 0; i < 26; i++) {
        if (freq[indices[i]] == 0) break;
        pushes += freq[indices[i]] * (1 + i / 8);
    }
    
    return pushes;
}